p_sets/3/generate.rb

```diff
@@ -0,0 +1,26 @@
+class RandomArrayGenerator
+  EXPONENT = 10**3
+  attr_reader :length, :seed
+
+  def initialize(length, seed = nil)
+    @length = length
+    @seed = seed
+  end
+
+  def array
+    Array.new(length) { (rand * EXPONENT).ceil }
+  end
+
+  def seed_rand
+    srand seed
```

Comment:
I think we prefer to always use brackets for parameterised function calls, right @gareth ? So `srand(seed)`.